Theoretical biology revision prep

drvinceknight opened the pull request.

drvinceknight committed:
Write code for figure showing learned SSE.

drvinceknight committed:
Address all of rev 1s comments and 1/2 of rev 2.

marcharper commented on main.tex:
Is that right? I think that was already known, the main contribution is being able to expand the definition of extortionate to arbitrary strategies, and then to show that this phenomenon still holds.

marcharper replied:
You say something similar around round line 575

drvinceknight replied:
Yes, you're right 👍

marcharper commented on main.tex:
```diff
@@ -102,19 +93,24 @@
+hilbe2015partners, ichinose2018zero, Moran1707} the true effectiveness of these
+strategies in an evolutionary setting was discussed. For
+example~\cite{adami2013evolutionary} showed that ZD strategies were not
+evolutionarily stable. Furthermore, in that work it was also postulated that
```
One of our ML papers also showed that evolutionary successful strategies can learn to self recognize. My "Art of War" paper shows that one can design strategies to do so, albeit not necessarily ZD strategies (no even memory one).

drvinceknight replied:
Will add these. Thank you.

marcharper commented on main.tex:
```diff
@@ -102,19 +93,24 @@
+of strategies are identified: partners and rivals and some discussion about the
+environments necessary for either to be evolutionary stable are given.
+
+In a non
```
need hyphen "non-evolutionary"

marcharper commented on main.tex:
```diff
@@ -173,33 +169,42 @@
 \section{Methods: Recognising Extortion}\label{sec:sserror-zd-strategies}
 
-Zero-determinant strategies are a special case of memory-one strategies,
+The role of this section is to review the definition of ZD
```
This section reviews ...., presents, ...

In general I'm not a fan of a section self-summarizing at the beginning, but perhaps this is responding to a specific comment

drvinceknight replied:
I'm not a fan at all either, this was specifically requested by reviewer 1.

marcharper commented on main.tex:
```diff
@@ -393,24 +399,29 @@ \subsection{Measuring Extortion from the History of Play}
 
 \section{Results: Validation of approach and Numerical experiments}\label{sec:numerical-experiments}
 
+The role of this section is to validate the approach of the previous section and
```
This section validates ...., and presents ...

marcharper commented on main.tex:
```diff
@@ -729,6 +765,25 @@ \section{Discussion}\label{sec:conclusion}
+There are various potential applications for \(\SSe\), one of which could be to
+devise an agent that learns during the interactions with another agent.
+Figure~\ref{fig:learning_sse} shows the average \(\SSe\) value over a number of
+iterations over a number of repetitions. More investigation would be required
```
Maybe rephrase what "would be interesting" to do to anything concrete we can say now. Otherwise the reviewers may ask that we do more of these ...

marcharper commented on reviews/2022-04-25.md:
```diff
@@ -1,16 +1,33 @@
+We believe we have improved the presentation of this and will discuss the
+modifications made in detail below.
+
+Reviewer 2s made 2 main suggestions for further work, we have added to the paper
```
2s --> 2

marcharper commented on reviews/2022-04-25.md:
```diff
@@ -97,9 +180,19 @@ clear presentation, interesing (although limited) idea and concrete simulation.
    Dilemma". Is it possible for this paper to handle the non-linear control
    cases that proposed in these papers？
+
+   TODO: Read to see if we can do this.
```
This is also out of scope IMO

marcharper commented:
Additional feedback:

* The plots in Figure 2 are too small, let's try a 2x2 instead of a 1x4 layout 
* Figure 3: do we want to add a best fit curve of some kind?
* Figure 8: Maybe expand the caption and call out ZD-SET-2 as generous

Nikoleta-v3 commented on reviews/2022-04-25.md:
```diff
@@ -1,16 +1,33 @@
+highlight that such claims are incomplete and in fact that extortion lacks the
+ability to adapt which is fundamentally important to evolution.
+
+We believe we have improved the presentation of this and will discuss the
```
It's not clear what `this` refers to. Merge this sentence with the previous paragraph?

drvinceknight replied:
Yup, thanks.

Nikoleta-v3 commented on reviews/2022-04-25.md:
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
+
+   The role of Section 2 is to give an overview of the mathematical definition
+   of Extortionate strategies so as to present the novel linear algebraic
+   measure of the our manuscript. We do not believe that we mainly focus on the
```
~the~

drvinceknight replied:
Actually I disagree, have left `... focus on the effectiveness` here. Let me know if I'm missing something.

Nikoleta-v3 replied:
Sorry my comment wasn't very clear, it took me a while to see it too. `measure of the our manuscript.` either remove `the` or `our`

drvinceknight replied:
Thank you!

Nikoleta-v3 commented on reviews/2022-04-25.md:
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
+
+   We have added the Hilbe et al. 2018 paper to our introduction in the
+   paragraph that had other papers that questioned the effectiveness of ZD
+   strategies in evolving populations. Thank you for this suggestion.
```
```suggestion
   We have added the Hilbe et al. 2018 paper to our introduction where we
   discuss  other papers that have questioned the effectiveness of ZD
   strategies in evolving populations. Thank you for this suggestion.
```

Nikoleta-v3 commented on reviews/2022-04-25.md:
```diff
@@ -30,48 +47,108 @@ immature in many senses. I raise the immature points below. Unless these issues
+   paragraph that had other papers that questioned the effectiveness of ZD
+   strategies in evolving populations. Thank you for this suggestion.
+
+   The 2013 work of Stewart and Plotkin is in the paper and in is the subject of
```
```suggestion
   The 2013 work of Stewart and Plotkin is in the paper and it is the subject of
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -93,6 +83,7 @@
     \label{eqn:theipd}
```
regarding the comment:

> The payoff matrix
> of PD in the Introduction and the replicator dynamics described as Eq. (25)
> are boorish because each variable is not explained anywhere.

Should we include more discussion regarding the payoff matrix? That `R` stand for the `Reward` payoff etc?

drvinceknight replied:
Yup! I think I meant to do this but forgot. Thanks.

Nikoleta-v3 commented on main.tex:
```diff
@@ -173,33 +169,42 @@
+The role of this section is to review the definition of ZD
+strategies from the literature, present the vector space in which such
+strategies exist and finally present a novel measure that allows for a measure
+of how far any memory one strategy is from the space of ZD
```
```suggestion
of how far any memory-one strategy is from the space of ZD
```

Nikoleta-v3 commented on main.tex:
```diff
@@ -547,23 +563,37 @@ \subsection{Numerical experiments}
+The main motivation for this consideration is to gain insights on to how
+behaviours might arise but also whether or not they are stable in various
+settings such as social and biological interactions. Most of such work
+considers the space of memory one strategies alone, in this paper a wider
```
```suggestion
considers the space of memory-one strategies alone, in this paper a wider
```

drvinceknight committed:
Apply suggestions from code review from Nik.

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Update main.tex

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Update main.tex

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Make some of the minor suggestions.

drvinceknight committed:
Make more adjustments based on review.

drvinceknight commented:
> The plots in Figure 2 are too small, let's try a 2x2 instead of a 1x4 layout

Done, I think it looks good, let me know what you think.

> Figure 3: do we want to add a best fit curve of some kind?

I've added a line. Played around with various other polynomials but I think the trend is the main thing that looked interesting. Happy to try something else though.

> Figure 8: Maybe expand the caption and call out ZD-SET-2 as generous

Done :)


I'll spend some time tomorrow tidying everything up but any other comments or things you'd like me to try please let me know :) 👍

drvinceknight committed:
Fix typo.

drvinceknight committed:
Merge branch 'theoretical-biology-revision-prep' of github.com:drvinceknight/testing_for_ZD into theoretical-biology-revision-prep

drvinceknight committed:
Make final amendment to discussion section.

drvinceknight committed:
Add a marked-up pdf.

drvinceknight commented:
I think I have modified everything and this is essentially good to go. I have added a marked-up pdf: https://github.com/drvinceknight/testing_for_ZD/blob/c00f8800a16b4c293b9d5dbdadbe9eeb2e3e65c7/diff.pdf

Let me know if you have any other thoughts (including if you think I could do a better job at anything you've already suggested).

marcharper commented:
The plots look much better. I opened a PR with some copy edits.

marcharper committed:
Some copy edits, mostly to the abstract and introduction (#130)

drvinceknight committed:
Recreate a diff'd pdf after @marcharper's edits.

drvinceknight commented:
> I opened a PR with some copy edits.

Thank you! I have pushed a new version of the diff'd pdf.

drvinceknight commented:
I'm going to go ahead and merge this. Will plan to resubmit on Monday. Thanks both 👍

Nikoleta-v3 commented:
Thank you @drvinceknight ! 🎉
